Cannot find module 'antlr4/index'

If I try to use the lib to compile a grammar outside of the source folder (e.g. on the Desktop), I'm receiving the following exception:

```
{ Error: Cannot find module 'antlr4/index'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:581:15)
    at Function.Module._resolveFilename (…/node_modules/tsconfig-paths/lib/register.js:73:40)
    at Function.Module._load (internal/modules/cjs/loader.js:507:25)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at sandbox.require.id (…/node_modules/node-eval/index.js:61:48)
    ***at Object.<anonymous> (…/Desktop/eval.js:3:14)*** 
    at _commonjsEval (…/node_modules/node-eval/index.js:80:21)
    at module.exports (…/node_modules/node-eval/index.js:36:19)
    at Object.readParser (…/node_modules/antlr4-tool/src/antlr-core/parser-util.ts:20:20)
    at AntlrCompiler.compileTypeScript (…/node_modules/antlr4-tool/src/antlr-core/antlr-compiler.ts:83:37)
    at …/node_modules/antlr4-tool/src/antlr-core/index.ts:48:63
    at …/node_modules/antlr4-tool/src/antlr-core/index.ts:20:25
    at arrayEach (…/node_modules/lodash/lodash.js:516:11)
    at Function.forEach (…/node_modules/lodash/lodash.js:9344:14)
    at compileWithFunction (…/node_modules/antlr4-tool/src/antlr-core/index.ts:10:7)
    at compileGrammarAsTypeScript (…/node_modules/antlr4-tool/src/antlr-core/index.ts:48:12)
```

--- a/src/antlr-core/parser-util.ts
+++ b/src/antlr-core/parser-util.ts
@@ -5,10 +5,16 @@ import * as util from './util';
 import * as vm from 'vm';
 
 export function readLexer(grammar: string, lexerFile: string) {
-    const contents = fs.readFileSync(lexerFile).toString();
-    const context = { require, exports: {} };
+    const code = fs.readFileSync(lexerFile).toString();
+    const vmRequire = (request: string) => {
+      if (request === 'antlr4/index' || request === 'antlr4') {
+        return require(request);
+      }
+      return require.resolve(request, { paths: [path.dirname(lexerFile)] });
+    };
+    const context = { require: vmRequire, exports: {} as any, __dirname: path.dirname(lexerFile), __filename: path.resolve(lexerFile) };
     vm.createContext(context);
-    vm.runInContext(contents, context);
+    vm.runInContext(code, context);
     const Lexer = context.exports[`${grammar}Lexer`];
     const lexer = new Lexer(null);
 
@@ -16,10 +22,16 @@ export function readLexer(grammar: string, lexerFile: string) {
 }
 
 export function readParser(grammar: string, parserFile: string) {
-    const contents = fs.readFileSync(parserFile).toString();
-    const context = { require, exports: {} };
+    const code = fs.readFileSync(parserFile).toString();
+    const vmRequire = (request: string) => {
+      if (request === 'antlr4/index' || request === 'antlr4') {
+        return require(request);
+      }
+      return require.resolve(request, { paths: [path.dirname(parserFile)] });
+    };
+    const context = { require: vmRequire, exports: {} as any, __dirname: path.dirname(parserFile), __filename: path.resolve(parserFile) };
     vm.createContext(context);
-    vm.runInContext(contents, context);
+    vm.runInContext(code, context);
     const Parser = context.exports[`${grammar}Parser`];
     const parser = new Parser(null);
 

--- a/src/antlr-core/parser-util.ts
+++ b/src/antlr-core/parser-util.ts
@@ -7,10 +7,10 @@ import * as vm from 'vm';
 export function readLexer(grammar: string, lexerFile: string) {
     const code = fs.readFileSync(lexerFile).toString();
     const vmRequire = (request: string) => {
-      if (request === 'antlr4/index' || request === 'antlr4') {
-        return require(request);
-      }
-      return require.resolve(request, { paths: [path.dirname(lexerFile)] });
+        if (request === 'antlr4/index' || request === 'antlr4') {
+            return require(request);
+        }
+        return require.resolve(request, { paths: [path.dirname(lexerFile)] });
     };
     const context = { require: vmRequire, exports: {} as any, __dirname: path.dirname(lexerFile), __filename: path.resolve(lexerFile) };
     vm.createContext(context);
@@ -24,10 +24,10 @@ export function readLexer(grammar: string, lexerFile: string) {
 export function readParser(grammar: string, parserFile: string) {
     const code = fs.readFileSync(parserFile).toString();
     const vmRequire = (request: string) => {
-      if (request === 'antlr4/index' || request === 'antlr4') {
-        return require(request);
-      }
-      return require.resolve(request, { paths: [path.dirname(parserFile)] });
+        if (request === 'antlr4/index' || request === 'antlr4') {
+            return require(request);
+        }
+        return require.resolve(request, { paths: [path.dirname(parserFile)] });
     };
     const context = { require: vmRequire, exports: {} as any, __dirname: path.dirname(parserFile), __filename: path.resolve(parserFile) };
     vm.createContext(context);

--- a/src/antlr-core/parser-util.ts
+++ b/src/antlr-core/parser-util.ts
@@ -2,22 +2,25 @@ import * as path from 'path';
 import * as fs from 'fs';
 import * as _ from 'lodash';
 import * as util from './util';
-
-const _eval = require('node-eval');
+import * as vm from 'vm';
 
 export function readLexer(grammar: string, lexerFile: string) {
-    const outputDir = path.dirname(lexerFile);
     const contents = fs.readFileSync(lexerFile).toString();
-    const Lexer = _eval(contents, `${outputDir}/eval.js`)[`${grammar}Lexer`];
+    const context = { require, exports: {} };
+    vm.createContext(context);
+    vm.runInContext(contents, context);
+    const Lexer = context.exports[`${grammar}Lexer`];
     const lexer = new Lexer(null);
 
     return lexer;
 }
 
 export function readParser(grammar: string, parserFile: string) {
-    const outputDir = path.dirname(parserFile);
     const contents = fs.readFileSync(parserFile).toString();
-    const Parser = _eval(contents, `${outputDir}/eval.js`)[`${grammar}Parser`];
+    const context = { require, exports: {} };
+    vm.createContext(context);
+    vm.runInContext(contents, context);
+    const Parser = context.exports[`${grammar}Parser`];
     const parser = new Parser(null);
 
     return parser;

